sdk/cosmosdb/cosmos/test/public/functional/dotnetcse.spec.ts

```diff
@@ -1400,6 +1372,94 @@ describe("dotnet test cases", () => {
     console.log(`unwrapCount for 2:2: ${unwrapCount}`);
     // assert.equal(unwrapCount, 2, "we have two cek");
   });
+
+  it("key encryption key revoke test", async () => {
+    const keyEncryptionKeyResolver = new MockKeyVaultEncryptionKeyResolver();
+    const encryptionTestClient = new CosmosClient({
+      endpoint: endpoint,
+      key: masterKey,
+      enableEncryption: true,
+      keyEncryptionKeyResolver: keyEncryptionKeyResolver,
+      encryptionKeyTimeToLiveInHours: 0,
+      encryptionKeyResolverName: testKeyVault,
+    });
+
+    // await removeAllDatabases();
+
+    const revokedKekMetadata = new EncryptionKeyWrapMetadata(
+      testKeyVault,
+      "revokedKek",
+      "revokedKek-metadata",
+      KeyEncryptionKeyAlgorithm.RSA_OAEP,
+    );
+    const testdatabase = (await encryptionTestClient.databases.createIfNotExists({ id: "myDb1" }))
+      .database;
+    await testdatabase.createClientEncryptionKey(
+      "keyWithRevokedKek",
+      EncryptionAlgorithm.AEAD_AES_256_CBC_HMAC_SHA256,
+      revokedKekMetadata,
+    );
+
+    // Once a Dek gets cached and the Kek is revoked, calls to unwrap/wrap keys would fail since KEK is revoked.
+    // The Dek should be rewrapped if the KEK is revoked.
+    // When an access to KeyVault fails, the Dek is fetched from the backend (force refresh to update the stale DEK) and cache is updated.
+    const pathWithRevokedKek = new ClientEncryptionIncludedPath(
+      "/sensitive_NestedObjectFormatL1",
+      "keyWithRevokedKek",
+      EncryptionType.DETERMINISTIC,
+      EncryptionAlgorithm.AEAD_AES_256_CBC_HMAC_SHA256,
+    );
+    const paths = [pathWithRevokedKek];
+    const clientEncryptionPolicyWithRevokedKek = new ClientEncryptionPolicy(paths);
+    const containerProperties = {
+      id: randomUUID(),
+      partitionKey: {
+        paths: ["/PK"],
+      },
+      clientEncryptionPolicy: clientEncryptionPolicyWithRevokedKek,
+    };
+    const testcontainer = (await testdatabase.containers.create(containerProperties)).container;
+    const testDoc1 = new TestDoc((await testCreateItem(testcontainer)).resource);
+    keyEncryptionKeyResolver.revokeAccessSet = true;
+    // Try creating it and it should fail as it has been revoked.
+    try {
+      await testCreateItem(testcontainer);
+      // await testcontainer.items.create({
```

Comment:
should this be uncommented?

Reply:
will remove the commented part. testCreateItem also creates the item and verifies the response